dockers/docker-dhcp-server/dhcpservd.py

```diff
@@ -0,0 +1,521 @@
+                    config_subnet = {
+                        "subnet": str(ipaddress.ip_network(dhcp_interface_ip, False)),
+                        "pools": [],
+                        "option-data": [
+                            {
+                                "name": "routers",
+                                "data": "{}".format(gateway)
+                            },
+                            {
+                                "name": "dhcp-server-identifier",
+                                "data": "{}".format(server_id)
+                            }
+                        ],
+                        "valid-lifetime": lease_time,
+                        "reservations": []
+                    }
+                    # Construct ip pools based on ip range.
+                    for port_name, ip_ranges in ports.items():
+                        if port_name in classes_port:
+                            class_set.append(client_class[port_name])
+                        for ip_range in ip_ranges:
+                            pool = {
+                                "pool": "{} - {}".format(str(ip_range[0]), str(ip_range[1])),
+                                "client-class": "{}:{}".format(hostname, port_name)
+                            }
+                            config_subnet["pools"].append(pool)
+                    # TODO Add customized options
+                    self.kea_config["Dhcp4"]["subnet4"].append(config_subnet)
+        if len(class_set) != 0:
+            self.kea_config["Dhcp4"]["client-classes"] = class_set
+
+        with open("/etc/kea/kea-dhcp4.conf", "w") as file:
+            json.dump(self.kea_config, file, indent=4, ensure_ascii=False)
+
+    def clear_table(self, db, table_name):
+        """
+        Clear table in db.
+        Args:
+            db: Db object.
+            table_name: Name of table to be clear
+        """
+        for key in db.keys("{}*".format(table_name)):
+            db.delete(key)
+
+    def update_lease_handler(self, signum, frame):
+        """
+        Hanlder function to update lease table in STATE_DB
+        """
+        # Read lease file generated by kea-dhcp4
+        lease_obj = {}
+        with open(LEASE_FILE_PATH, "r", encoding="utf-8") as fb:
+            dq = deque(fb)
+        while dq:
+            last_row = dq.pop()
+            splits = last_row.split(",")
+            if splits[0] == "address":
+                break
+            ip_str = splits[0]
+            mac_address = splits[1]
+            if mac_address in lease_obj:
+                continue
+            valid_lifetime = splits[3]
+            lease_end = splits[4]
+            lease_obj[mac_address] = {
+                "lease_start": str(int(lease_end) - int(valid_lifetime)),
+                "lease_end": lease_end,
+                "ip": ip_str
+            }
+        # Clear lease table in STATE_DB
```

Comment:
Clear will cause all the leases lost. 
Please do incremental update instead:
1. Find what need to be add/update, and update entry
2. Find what need to be remove, and remove entry